terror/terror.go

```diff
@@ -360,6 +361,8 @@ func Call(fn func() error) {
 // Log logs the error if it is not nil.
 func Log(err error) {
 	if err != nil {
-		log.Error("encountered error", zap.Error(err))
+		var buf [4096]byte
+		sz := runtime.Stack(buf[:], false)
+		log.Error("encountered error", zap.Error(err), zap.String("stack", string(buf[:sz])))
```

Comment:
Perhaps easier to just do

```suggestion
		log.Error("encountered error", zap.Error(errors.Trace(err)))
```

Reply:
Yes, we need it!
We need the call stack to identify the misuse of `terror.Log()`.  @kennytm

Reply:
@tiancaiamao Ah ok. In that case `errors.Trace` isn't suitable since it won't replace the error's stack if it already exists. We should use [`errors.WithStack`](https://pkg.go.dev/github.com/pingcap/errors?tab=doc#WithStack)

Reply:
Done